7/src/part1.c

```diff
@@ -0,0 +1,175 @@
+// Link: https://adventofcode.com/2022/day/7
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define MAX_CONTENTS 100
+
+
+typedef struct file_or_dir {
+    char name[256];
+    int is_directory;                            // 1 if directory, 0 if file
+    int size;                                    // size of the directory or file
+    struct file_or_dir* contents[MAX_CONTENTS];  // children if it's a directory
+    struct file_or_dir* parent;                  // pointer to the parent directory
+} file_or_dir;
+
+file_or_dir* root;
+
+
+file_or_dir* create_directory(const char* name) {
+    file_or_dir* new_dir = malloc(sizeof(file_or_dir));
+    strcpy(new_dir->name, name);
+    new_dir->is_directory = 1;
+    new_dir->size = 0;
+    new_dir->parent = NULL;
+    for (int i = 0; i < MAX_CONTENTS; i++) {
+        new_dir->contents[i] = NULL;
+    }
+    return new_dir;
+}
+
+file_or_dir* create_file(const char* name, int size) {
+    file_or_dir* new_file = malloc(sizeof(file_or_dir));
+    strcpy(new_file->name, name);
+    new_file->is_directory = 0;
+    new_file->size = size;
+    new_file->parent = NULL;
+    for (int i = 0; i < MAX_CONTENTS; i++) {
+        new_file->contents[i] = NULL;
+    }
+    return new_file;
+}
+
+
+void add_to_directory(file_or_dir* parent, file_or_dir* child) {
```

Comment:
add realloc of memory here 
This reduces the memory overhead and makes the code more scalable.
    if(parent->content_count % 10 == 0) {
        parent->contents = realloc(parent->contents, sizeof(file_or_dir*) * (parent->content_count + 10));
    }